data_type/data_optimization.ipynb

```diff
@@ -0,0 +1,2756 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f505c841-eba0-4a65-826b-529bd51d090d",
+   "metadata": {},
+   "source": [
+    "# Optimizing data access and processing\n",
+    "\n",
+    "Over the multiple formats presented in the other tutorials, some were presented as cloud optimized. This notebook shows some methods to take advantage of those formats to optimize data access and processing. The methods presented are specifically optimized for cloud usage but can (and should when needed) also be used with local files. The general idea is to limit the data access: either by loading only the required data or opening small fractions (chunks) of a whole dataset and iterate over all these chunks.\n",
+    "\n",
+    "Some examples in this notebook use data downloaded or generated in other notebooks. Make sure you ran them before running this one. They also provide useful information and context on the files and their formats.\n",
+    "\n",
+    "\n",
+    "## Raster data\n",
+    "\n",
+    "In the [raster formats tutorial](./raster_formats.ipynb), COGs (cloud optimized geotiffs) were presented as the best cloud-oriented solution, as they are natively split in chunks and allow overviews. In this section, we present how to load and process a raster chunk by chunk. This is particularly usefull when using large datasets which can't be loaded in memory (both on local or cloud architectures).\n",
+    "\n",
+    "### Using rasterio\n",
+    "\n",
+    "Reading chunks with rasterio is done using a window: a smaller section of the image, defined by an offset and a size on the x and y axes. A simple way to implement reading an image chunk by chunk is using chunks with the same width as the image (i.e. one or multiple lines) and iterating over them."
```

Comment:
It is a bit weird to first show how to read by lines when we speak about a tiled dataset, don't you think?